DigitalPulseSoftware/BurgWar
# [WIP] Burgwar Docker image - Xmake compatible version

This is the first working version of Docker image.

# How to test it ?

- `git clone` the folder
- Build image with `docker build --tag burgwar-server:latest .`
- Run server:
```
docker run -ti --rm --name=burgwar-server -p 14768:14768/udp \
 -v /relativeOrAbsolute/path/to/assets:/srv/assets/ \
 -v /relativeOrAbsolute/path/to/mods:/srv/mods/ \
 -v /relativeOrAbsolute/path/to/scripts:/srv/scripts/ \
 -v /absolute/path/to/serverconfig.lua:/srv/serverconfig.lua \
 burgwar-server:latest
```
**Warning:** If you are using Docker Desktop on Windows, you should execute this command under a WSL console. Executing with Git bash for example won't work.

# TO DO before merge

- [x] use /home/burgwar instead of /home/burgwar/BurgWar
- [x] use default xmake.lua
- [x] Only keep artifacts to make a light image
- [x] Set server path in runtime image to /srv/
- [x] stop using git clone and use local repo ?

# Next steps
- Use alpine linux instead of debian for runtime image (#65)
- CI (#67)
- remove nazara legacy apt when we migrated (#66)
